integration-tests/src/test/java/oracle/kubernetes/operator/ITSitConfig.java

```diff
@@ -0,0 +1,277 @@
+            domain.getDomainNS());
+    Assert.assertFalse(stdout.toLowerCase().contains("error"));
+    testCompletedSuccessfully = true;
+    logger.log(Level.INFO, "SUCCESS - {0}", testMethod);
+  }
+
+  /**
+   * This test covers custom situational configuration use cases for diagnostics resource It adds a
+   * bunch of instrumentation monitors and harvesters in a diagnostics module.
+   *
+   * @throws Exception
+   */
+  @Test
+  public void testCustomSitConfigOverridesForWldf() throws Exception {
+    Assume.assumeFalse(QUICKTEST);
+    boolean testCompletedSuccessfully = false;
+    String testMethod = new Object() {}.getClass().getEnclosingMethod().getName();
+    logTestBegin(testMethod);
+    String stdout =
+        callShellScriptByExecToPod(
+            "runSitConfigTests.sh",
+            fqdn + " " + T3CHANNELPORT + " weblogic welcome1 " + testMethod,
+            ADMINPODNAME,
+            domain.getDomainNS());
+    Assert.assertFalse(stdout.toLowerCase().contains("error"));
+    testCompletedSuccessfully = true;
+    logger.log(Level.INFO, "SUCCESS - {0}", testMethod);
+  }
+
+  private static Domain createSitConfigDomain() throws Exception {
+    String createDomainScript = TESTSCRIPTDIR + "/domain-home-on-pv/create-domain.py";
+    // load input yaml to map and add configOverrides
+    Map<String, Object> domainMap = TestUtils.loadYaml(DOMAINONPVWLSTFILE);
+    domainMap.put("configOverrides", "sitconfigcm");
+    domainMap.put(
+        "configOverridesFile", "/integration-tests/src/test/resources/sitconfig/configoverrides");
+    domainMap.put("domainUID", DOMAINUID);
+    domainMap.put("adminNodePort", new Integer(ADMINPORT));
+    domainMap.put("t3ChannelPort", new Integer(T3CHANNELPORT));
+    domainMap.put(
+        "createDomainPyScript",
+        "integration-tests/src/test/resources/sitconfig/scripts/create-domain-auto-custom-sit-config20.py");
+
+    // use NFS for this domain on Jenkins, defaultis HOST_PATH
+    if (System.getenv("JENKINS") != null && System.getenv("JENKINS").equalsIgnoreCase("true")) {
+      domainMap.put("weblogicDomainStorageType", "NFS");
+    }
+    domain = TestUtils.createDomain(domainMap);
+    domain.verifyDomainCreated();
+    return domain;
+  }
+
+  private static void destroySitConfigDomain() throws Exception {
+    if (domain != null) {
+      domain.destroy();
+    }
+  }
+
+  private static String callShellScriptByExecToPod(
+      String scriptPath, String arguments, String podName, String namespace) throws Exception {
+
+    StringBuffer cmdKubectlSh = new StringBuffer("kubectl -n ");
+    cmdKubectlSh
+        .append(namespace)
+        .append(" exec -it ")
+        .append(podName)
+        .append(" -- bash -c 'sh ")
+        .append(scriptPath)
+        .append(" ")
+        .append(arguments)
+        .append("'");
+    logger.info("Command to call kubectl sh file " + cmdKubectlSh);
+    ExecResult result = ExecCommand.exec(cmdKubectlSh.toString());
```

Comment:
check TestUtils.exec(str) method if you can replace the below lines with that.

Reply:
The reason why I am using my own method is the TestUtils.exec(str) doesn't dump stdout when there is a error. I would also to printout stdout in case of a error

Reply:
It does, it prints in the exception message.  
```
      throw new RuntimeException(
          "FAILURE: Command " + cmd + " failed with " + result.stderr() + " \n " + result.stdout());
```

Reply:
i agree you should use TestUtils.exec(str)

Reply:
I've "unresolved" this conversation because it seems that Vanaja and Mark's concern wasn't addressed.

Reply:
Regardless please keep in mind that it's helpful to include _both_ the stdout and stderr in an 'INFO' message _in addition to_ throwing a runtime which also has _both_ the stdout and stderr.   The code here is not doing this with the INFO message.    The reason is that (a) an INFO is helpful as the runtime isn't logged at the same location the test is run, which makes reading the logs confusing, and (b) it isn't predictable whether tracing/errors are going to be in stdout or stderr.

Reply:
Fixed to use TestUtils.exec(str)